app/models/user.rb

```diff
@@ -3,4 +3,11 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  has_many :favorites, dependent: :destroy
+  has_many :posts, through: :favorites
+
+  def full_name
+    [first_name, last_name].compact.join(' ')
+  end
```

Comment:
I use `.compact.join(' ')` so users without either `first_name` or `last_name` are still valid (no validation against not having either a first or last name)